sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/PartialMethodNoImplementation.cs

```diff
@@ -57,8 +57,10 @@ private static void CheckForCandidatePartialDeclaration(SyntaxNodeAnalysisContex
         {
             var declaration = (MethodDeclarationSyntax)context.Node;
             var partialKeyword = declaration.Modifiers.FirstOrDefault(m => m.IsKind(SyntaxKind.PartialKeyword));
-            if (declaration.Body != null ||
-                partialKeyword == default(SyntaxToken))
+
+            if (partialKeyword == default(SyntaxToken)||
+                declaration.Body != null ||
+                declaration.ExpressionBody != null)

```

Comment:
It was working without the new check but I think that's better to do early return whenever possible.